src/Db/Service/ToggleStorage.php

```diff
@@ -0,0 +1,225 @@
+<?php
+namespace Clearbooks\Labs\Db\Service;
+
+use Clearbooks\Labs\Db\Table\Toggle as ToggleTable;
+use Clearbooks\Labs\Db\Table\UserPolicy as UserPolicyTable;
+use Clearbooks\Labs\Db\Table\GroupPolicy as GroupPolicyTable;
+use Clearbooks\Labs\Db\Entity\Toggle;
+use Clearbooks\Labs\Toggle\UseCase\GroupPolicyRetriever;
+use Clearbooks\Labs\Toggle\UseCase\ToggleRetriever;
+use Clearbooks\Labs\Toggle\UseCase\UserPolicyRetriever;
+use Doctrine\DBAL\Connection;
+
+class ToggleStorage implements ToggleRetriever, UserPolicyRetriever, GroupPolicyRetriever
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var ToggleTable
+     */
+    private $toggleTable;
+
+    /**
+     * @var UserPolicyTable
+     */
+    private $userPolicyTable;
+
+    /**
+     * @var GroupPolicyTable
+     */
+    private $groupPolicyTable;
+
+    /**
+     * @param Connection       $connection
+     * @param ToggleTable      $toggleTable
+     * @param UserPolicyTable  $userPolicyTable
+     * @param GroupPolicyTable $groupPolicyTable
+     */
+    public function __construct( Connection $connection, ToggleTable $toggleTable, UserPolicyTable $userPolicyTable,
+                                 GroupPolicyTable $groupPolicyTable )
+    {
+        $this->connection = $connection;
+        $this->toggleTable = $toggleTable;
+        $this->userPolicyTable = $userPolicyTable;
+        $this->groupPolicyTable = $groupPolicyTable;
+    }
+
+    /**
+     * @param int $toggleId
+     * @return Toggle|null
+     */
+    public function getToggleById( $toggleId )
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
+        $queryBuilder->select( "*" )->from( (string)$this->toggleTable )->where( "id = ?" );
+        $queryBuilder->setParameter( 0, $toggleId );
+
+        $toggleData = $queryBuilder->execute()->fetch();
+        return !empty( $toggleData ) ? new Toggle( $toggleData ) : null;
+    }
+
+    /**
+     * @param Toggle $toggle
+     * @return int
+     */
+    public function insertToggle( Toggle $toggle )
```

Comment:
insertToggle only available here in ToggleStorage
but probably they will hack this :)